packages/icons/src/icon/index.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import { cloneElement, forwardRef } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+import type { ReactElement, ForwardedRef } from 'react';
+import type { SVGProps } from '@wordpress/primitives';
+
+export interface IconProps extends SVGProps {
+	icon: ReactElement;
+	size?: number;
+}
```

Comment:
We can add the docs for props here for better IDE integration

```suggestion
export interface IconProps extends SVGProps {
	/**
	 * The SVG component to render
	 */
	icon: React.ReactElement;

	/**
	 * The size of the icon in pixels
	 *
	 * @default 24
	 */
	size?: number;
}
```